continuous-testing/starter-kits/README.md

```diff
@@ -2,6 +2,403 @@
+
+3. Install pre-commit hook
+```bash
+pre-commit install
+```
+This command reads the `.pre-commit-config.yaml` file and installs the pre-commit hook in your local git repository.
+After this, you can see a `.git/hooks/pre-commit` file is created in your local git repository.
+
+4. Make sure you have the baseline file `.secrets.baseline` in your local git repository
+
+5. Commit your changes
+
+Now, you can commit your changes as usual. If any **new secrets** are detected, the commit will be prevented and the secrets will be reported.
+
+For example,
+
+<img width="559" alt="Screen Shot 2023-04-20 at 7 32 10 AM" src="https://user-images.githubusercontent.com/92573736/233398613-6e189322-4d97-47c3-b3ba-bd700a716cf6.png">
+
+
+> **Note**: pre-commit hook block commit by comparing new secrets with the results in `.secrets.baseline` file. If you want to add new secret results, you need to update `.secrets.baseline` file by re-running the scan command and generate a new baseline file.
+>
+> You can create an empty result baseline file by running this command at a directory without secrets.
+
+#### Layer 3: Server-side Push to GitHub.com
+The final layer of our solution is a server-side pre-commit scan powered by [GitHub Action](https://github.com/features/actions). This scan is triggered whenever a developer pushes to a branch or creates a pull request. If the scan detects any new secrets, it can generate a not detailed report compared to layer 2 (for security concern), email to the developer, and report a status check to GitHub. The status check will prevent the developer from merging the pull request or pushing to the **protected** branch. This layer protects the protected branch from being polluted by secrets, but secrets can still be pushed to other branches.
+
+```mermaid
+sequenceDiagram
+    participant User as Developer
+    participant GH as GitHub
+    participant Workflow as detect-secrets.yaml
+    participant GA as GitHub Action
+    participant DS as Detect-Secrets
+
+    Note over User,GH: Developer creates pull request or pushes to branch
+    User->>+GH: Creates pull request / pushes to branch
+    GH->>+Workflow: Triggers GitHub Action workflow
+    Workflow->>GA: Sets up and runs Detect-Secrets scan
+    GA->>DS: Requests secret scan
+    DS->>DS: Scans repository for secrets
+    alt Secrets Detected
+        DS-->>GA: Returns detected secrets
+        GA-->>GH: Fails status check
+        GH-->>User: Prevents merge / push & sends email notification
+    else No Secrets Detected
+        DS-->>GA: Returns clean result
+        GA-->>GH: Passes status check
+        GH-->>User: Allows merge / push
+    end
+
+```
+Starter Kit:
+1. Create a workflow file `detect-secrets.yaml` in `.github/workflows` directory from your repository root.
```

Comment:
One more note: try pushing this workflow to https://github.com/marketplace?type=actions when you are ready. This will make installation a breeze.

Reply:
When I publish it to the Marketplace, who should be the owner of this Action?
When I chose NASA-AMMOS, it asked me to contact the admin of the organization to accept the Marketplace agreement:
<img width="1165" alt="image" src="https://github.com/NASA-AMMOS/slim/assets/92573736/1c12cb15-5b5f-4e14-8287-2005e2747330">

If I publish it by own account, I can accept the agreement anyway.

Reply:
Currently, I published my GitHub Action with additional documentation through my account. 
Here is the [link](https://github.com/marketplace/actions/detect-secrets-action) to it.

Reply:
@perryzjc - its fine to keep on your account for now, until we come up with a better organization contact info.